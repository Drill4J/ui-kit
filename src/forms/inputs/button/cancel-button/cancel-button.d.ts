/// <reference types="react" />
interface Props {
    className?: string;
    children: React.ReactNode;
    onClick?: () => void;
    size?: 'large' | 'small';
    disabled?: boolean;
    'data-test'?: string;
}
export declare const CancelButton: ({ className, children, ...rest }: Props) => JSX.Element;
export {};
