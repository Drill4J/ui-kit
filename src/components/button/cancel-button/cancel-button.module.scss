@import '../../../style-guide/index';

.cancel-button {
  border: 1px solid $loblolly-color;
  background-color: map-get($color-palette, white);
  color: $slate-gray;
  font-family: $semibold-font;

  &:hover {
    border: 1px solid map-get($color-palette, black10);
    color: $shade-gray;
  }

  &:active {
    border: 1px solid $shade-gray;
    color: $shade-gray;
  }
}
