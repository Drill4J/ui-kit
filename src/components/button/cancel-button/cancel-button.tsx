import * as React from 'react';
import { BEM } from '@redneckz/react-bem-helper';

import { Button } from '../button';
import { ButtonType } from '../button-type';

import styles from './cancel-button.module.scss';

interface Props extends ButtonType {
  className?: string;
  children: React.ReactNode;
  onClick?: () => void;
}

const cancelButton = BEM(styles);

export const CancelButton = cancelButton(({ className, children, ...rest }: Props) => (
  <Button className={className} {...rest}>
    {children}
  </Button>
));
