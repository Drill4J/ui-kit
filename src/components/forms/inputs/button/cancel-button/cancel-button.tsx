import tw, { styled } from 'twin.macro';

import { Button } from '../button';

export const CancelButton = styled(Button)`
  ${tw`
    border border-red-default bg-monochrome-white text-red-default font-bold
    hover:border-red-medium-tint hover:text-red-medium-tint
    active:border-red-shade active:text-red-shade
  `};
`;
