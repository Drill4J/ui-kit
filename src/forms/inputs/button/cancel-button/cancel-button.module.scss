@import '../../../../style-guide/index';

.cancel-button {
  border: 1px solid map-get($red, default);
  background-color: map-get($monochrome, white);
  color: map-get($red, default);
  font-family: $semibold-font;

  &:hover {
    border: 1px solid map-get($red, medium-tint);
    color: map-get($red, medium-tint);
  }

  &:active {
    border: 1px solid map-get($red, shade);
    color: map-get($red, shade);
  }
}
