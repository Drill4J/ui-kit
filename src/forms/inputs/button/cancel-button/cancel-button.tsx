import styled from 'styled-components';

import { Button } from '../button';
import { COLORS, FONTS } from '../../../../theme';

export const CancelButton = styled(Button)`
  border: 1px solid ${COLORS.RED.DEFAULT};
  background-color: ${COLORS.MONOCHROME.WHITE};
  color: ${COLORS.RED.DEFAULT};
  font-family: ${FONTS.SEMI_BOLD};

  &:hover {
    border: 1px solid ${COLORS.RED.MEDIUM_TINT};
    color: ${COLORS.RED.MEDIUM_TINT};
  }

  &:active {
    border: 1px solid ${COLORS.RED.SHADE};
    color: ${COLORS.RED.SHADE};
  }
`;
