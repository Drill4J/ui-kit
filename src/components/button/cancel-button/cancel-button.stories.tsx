import React from 'react';
import { storiesOf } from '@storybook/react';

import { CancelButton } from './cancel-button';

storiesOf('CancelButton', module)
  .add('Primary large', () => <CancelButton size="large">Primary large</CancelButton>)
  .add('Secondary small', () => <CancelButton size="small">Secondary small</CancelButton>);
