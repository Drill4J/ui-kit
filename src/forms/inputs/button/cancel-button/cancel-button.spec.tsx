import React from 'react';
import renderer from 'react-test-renderer';

import { CancelButton } from './cancel-button';

it('should match snapshot', () => {
  const tree = renderer.create(<CancelButton size="large">Cancel button</CancelButton>).toJSON();

  expect(tree).toMatchSnapshot();
});
