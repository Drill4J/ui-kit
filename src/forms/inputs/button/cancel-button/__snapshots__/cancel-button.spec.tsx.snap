// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  className="button button--size--large cancel-button"
  data-test=""
  disabled={false}
  onClick={[Function]}
  size="large"
  type="primary"
>
  Cancel button
</button>
`;
